lib/chat_app/contexts/rooms.ex

```diff
@@ -6,9 +6,16 @@ defmodule ChatApp.Contexts.Rooms do
 
   def get_room(id), do: Repo.get(Room, id)
 
-  def get_room_messages(id), do: Repo.get(Room, id) |> Repo.preload([:messages])
+  def get_room_messages(id) do
+    room = Repo.get(Room, id) |> Repo.preload([:messages])
+    comparator = fn x, y -> DateTime.compare(x.inserted_at, y.inserted_at) == :gt end
+    Enum.sort(room.messages, comparator)
+  end
 
-  def get_room_users(id), do: Repo.get(Room, id) |> Repo.preload([:users])
+  def get_room_users(id) do
+    room = Repo.get(Room, id) |> Repo.preload([:users])
+    room.users
+  end
```

Comment:
What if the room of the given ID does not exist?

Reply:
Good point, maybe that case can be handled by using pattern-matching. I'll fix it.